src/sidebar/components/annotation-body.js

```diff
@@ -11,8 +11,24 @@ import MarkdownView from './markdown-view';
 import TagEditor from './tag-editor';
 import TagList from './tag-list';
 
+/** @typedef {import("../../types/api").Annotation} Annotation */
+
+/**
+ * @typedef AnnotationBodyProps
+ * @prop {Annotation} props.annotation - The annotation in question
+ * @prop {boolean} [props.isEditing] - Whether to display the body in edit mode (if true) or view mode.
+ * @prop {(a: Object<'tags', string[]>) => void} [props.onEditTags] - Callback invoked when the user edits tags.
+ * @prop {(a?: Object<'text', string>) => void} [props.onEditText] - Callback invoked when the user edits the content of the annotation body.
```

Comment:
The types for `onEditTags` and `onEditText` look weird to me. You'd expect the callback arguments to be just a list of strings and a string respectively. Then I noticed a comment in one of the other modules that said "simplify this after we finish the Preact conversion", so this is just documenting the weirdness accurately for now.

Reply:
so its not a simple string or list of strings, for some reason or another. I didn't dig into this, but perhaps there is a opportunity to simplifying this down

If you look at tag-editor.js, the current code is 

`onEditTags({ tags: tagList });`

but perhaps it could just be 
`onEditTags(tagList);`

with a few simple edits.